### PR TITLE
Fix Admin troubleshooter tool

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?

Fetch loadbalancers info (url, ...) right after app startup.
Handle missing elb properly.
Check via https
## What is the value of this and can you measure success?

Ensure the Troubleshoot tool (https://frontend.gutools.co.uk/troubleshoot/pages) works all the time
## Request for comment

@jfsoul @guardian/dotcom-platform

### CODE REVIEW

File: admin/app/controllers/TroubleshooterController.scala [hunk 1]
```diff
@@ -30,18 +30,14 @@ class TroubleshooterController(wsClient: WSClient) extends Controller with Loggi
 
     val loadBalancers = LoadBalancer.all.filter(_.testPath.isDefined)
 
-    val thisLoadBalancer = loadBalancers.find(_.project == id).head
+    val thisLoadBalancer = loadBalancers.find(_.project == id).headOption
 
+    val directToLoadBalancer = thisLoadBalancer.map(testOnLoadBalancer(_, testPath, id))
+      .getOrElse(Future.successful(TestFailed("Can find the appropriate loadbalancer")))
```
Comment (jfsoul) [inline admin/app/controllers/TroubleshooterController.scala]: Can't?

File: admin/app/controllers/TroubleshooterController.scala [hunk 2]
```diff
@@ -60,19 +56,30 @@ class TroubleshooterController(wsClient: WSClient) extends Controller with Loggi
...
+          TestFailed(result.name, result.messages.toSeq :+
+            "NOTE: if hitting the Router you MUST set Host header to 'www.theguardian.com' or else you will get '403 Forbidden'":_*)
+      }
     }
+
+    LoadBalancer.all
+      .find(_.project == "frontend-router")
+      .flatMap(_.url)
+      .map(fetchWithRouterUrl(_))
+      .getOrElse(Future.successful(TestFailed("Can get Frontend router url")))
```
Comment (jfsoul) [inline admin/app/controllers/TroubleshooterController.scala]: Can't?

File: admin/app/controllers/TroubleshooterController.scala [hunk 3]
```diff
@@ -60,19 +56,30 @@ class TroubleshooterController(wsClient: WSClient) extends Controller with Loggi
...
+
   }
 
   private def testOnLoadBalancer(thisLoadBalancer: LoadBalancer, testPath: String, id: String): Future[EndpointStatus] = {
-    httpGet(s"Can fetch directly from ${thisLoadBalancer.name} load balancer", s"http://${thisLoadBalancer.url.head}$testPath")
+    thisLoadBalancer.url.map { url =>
+      httpGet(s"Can fetch directly from ${thisLoadBalancer.name} load balancer", s"http://${url}$testPath")
+    }.getOrElse(Future(TestFailed(s"Can get ${thisLoadBalancer.name}'s loadbalancer url")))
```
Comment (jfsoul) [inline admin/app/controllers/TroubleshooterController.scala]: Can't?

**Review by jfsoul — APPROVED** [review]

File: admin/app/views/troubleshooterResults.scala.html [hunk 4]
```diff
@@ -1,9 +1,9 @@
-@(loadbalancer: tools.LoadBalancer, results: Seq[controllers.admin.EndpointStatus])(implicit request: RequestHeader)
+@(loadbalancer: Option[tools.LoadBalancer], results: Seq[controllers.admin.EndpointStatus])(implicit request: RequestHeader)
 
 @admin_main("Troubleshooting results", "PROD", isAuthed = true) {
 
     <div class="row-fluid">
-        <h4>@loadbalancer.name (@loadbalancer.project)</h4>
+        <h4>@loadbalancer.map(lb => s"${lb.name} (${lb.project})").getOrElse("Unknown loadbalancer")</h4>
```
Comment (jfsoul) [inline admin/app/views/troubleshooterResults.scala.html]: What was wrong with fold?

Comment (TBonnin) [inline admin/app/views/troubleshooterResults.scala.html]: Consistency. I use map/getOrElse everywhere else in this PR

File: admin/app/controllers/TroubleshooterController.scala (hunk 1 above)
Comment (TBonnin) [inline admin/app/controllers/TroubleshooterController.scala]: Checks are reported like this
screen shot 2016-09-29 at 12 20 46

File: admin/app/views/troubleshooterResults.scala.html (hunk 4 above)
Comment (rjmk) [inline admin/app/views/troubleshooterResults.scala.html]: Fold is way cooler though 😎

**Pull request merged**